rts/Lua/LuaMathExtra.cpp

```diff
@@ -184,6 +186,42 @@ int LuaMathExtra::smoothstep(lua_State* L) {
 	return 1;
 }
 
+
+/*** Returns the length of an given vector.
+ *
+ * @function math.vectorlength
+ * @number x
+ * @number y
+ * @number z
+ * @treturn vector length
+ */
+int LuaMathExtra::vectorlength(lua_State* L)
+{
+	RECOIL_DETAILED_TRACY_ZONE;
+	float3 float3(luaL_checkfloat(L, 1), luaL_checkfloat(L, 2), luaL_checkfloat(L, 3));
+	lua_pushnumber(L, float3.Length());
+	return 1;
+}
+
+/*** Returns the normalize vector of an given vector.
+ *
+ * @function math.vectorvector
+ * @number x
+ * @number y
+ * @number z
+ * @treturn normalize vector
+ */
+int LuaMathExtra::normalizevector(lua_State* L)
```

Comment:
Should handle any dimensional vector, not just 3.

Reply:
Done